packages/fluentui/docs/src/examples/components/Menu/Visual/MenuExampleCompose.shorthand.tsx

```diff
@@ -19,32 +23,80 @@ const MenuItemIconGreen = compose(MenuItemIcon, {
   displayName: 'MenuItemIconGreen',
 });
 
-const MenuItemContentPink = compose(MenuItemContent, {
-  displayName: 'MenuItemContentPink',
+const MenuItemContentPurple = compose(MenuItemContent, {
+  displayName: 'MenuItemContentPurple',
 });
 
 const MenuitemIndicatorSaturated = compose(MenuItemIndicator, {
   displayName: 'MenuitemIndicatorSaturated',
 });
 
-const MenuItemBlue = compose(MenuItem, {
-  displayName: 'MenuItemBlue',
+interface MenuItemBlueOwnProps {
+  level?: number;
+}
+
+interface MenuItemBlueProps extends MenuItemProps, MenuItemBlueOwnProps {}
+
+const MenuItemBlue = compose<'a', MenuItemBlueProps, MenuItemStylesProps, MenuItemProps, MenuItemStylesProps>(
+  MenuItem,
+  {
+    displayName: 'MenuItemBlue',
+    slots: {
+      icon: MenuItemIconGreen,
+      content: MenuItemContentPurple,
+      indicator: MenuitemIndicatorSaturated,
+      wrapper: MenuItemWrapperDashed,
+    },
+    mapPropsToSlotProps: (props: MenuItemProps & { level?: number }) => ({
+      menu: {
+        level: props.level + 1,
+      },
+    }),
+  },
+);
+
+interface MenuColorfulOwnProps {
+  level?: number;
+}
+
+interface MenuColorfulProps extends MenuColorfulOwnProps, MenuProps {}
+
+interface MenuColorfulOwnStylesProps {
+  level?: number;
+}
+
+interface MenuColorfulStylesProps extends MenuColorfulOwnStylesProps, MenuStylesProps {}
+
+const MenuColorful = compose<'ul', MenuColorfulProps, MenuColorfulStylesProps, MenuProps, MenuStylesProps>(Menu, {
+  displayName: 'MenuColorful',
   slots: {
-    icon: MenuItemIconGreen,
-    content: MenuItemContentPink,
-    indicator: MenuitemIndicatorSaturated,
-    wrapper: MenuItemWrapperDashed,
+    item: MenuItemBlue,
   },
+  mapPropsToStylesProps: props => ({
+    level: props.level,
+  }),
+  mapPropsToSlotProps: props => ({
+    item: {
+      level: props.level || 0,
+    },
+  }),
 });
 
 const items = [
   {
     key: 'editorials',
     content: 'Editorials',
     icon: <BookmarkIcon />,
-    menu: ['One', 'Two', 'Three'],
-    // TODO: remove once compose is implemented for the hierarchical structure
-    children: (C, p) => <MenuItemBlue {...p} />,
+    defaultMenuOpen: true,
+    menu: [
+      {
+        key: 'One',
+        content: 'One',
+        menu: ['Four', { content: 'Five' }],
```

Comment:
```suggestion
        menu: { items: [{ content: 'Four' }, { content: 'Five' }], level: 1 },
```
???

Reply:
I want to have this in order for us to easily catch if the `shorthandConfig` is failing for some reason.